packages/app-bridge-app/src/AppBridgePlatformApp.ts

```diff
@@ -16,19 +15,47 @@ import {
     type StateReturn,
     type SubscribeMap,
 } from '@frontify/app-bridge';
+import { type Simplify } from 'type-fest';
 
 import { InitializationError } from './errors';
 import { type ApiMethodRegistry } from './registries';
 import { openConnection } from './registries/commands.ts';
+import { type ApiVerb } from './registries/verbs';
 import { Topic } from './types/Topic';
 import { generateRandomString, notify, subscribe } from './utilities';
 import { ErrorMessageBus, type IMessageBus, MessageBus } from './utilities/MessageBus';
 import { getQueryParameters } from './utilities/queryParams';
 
-export type PlatformAppApiMethod = ApiMethodNameValidator<
+type NameContextList = 'API Method';
+type WrongNamePattern<ApiMethodName, NameContext extends NameContextList> = ApiMethodName extends string
+    ? `The following ${NameContext} do not match the naming pattern: ${ApiMethodName}`
+    : never;
+
+type ObjectNameValidator<
+    NameObject,
+    PatternObject,
+    NameContext extends NameContextList,
+> = keyof NameObject extends keyof PatternObject
+    ? NameObject
+    : WrongNamePattern<
+          `${Exclude<Extract<keyof NameObject, string>, Extract<keyof PatternObject, string>>}`,
+          NameContext
+      >;
+
+type ApiMethodNamePattern = { [apiMethod: `${ApiVerb}${string}`]: { payload: unknown; response: unknown } };
+export type PlatformAppApiMethodNameValidator<ApiMethodNameObject> = Simplify<
+    ObjectNameValidator<ApiMethodNameObject, ApiMethodNamePattern, 'API Method'>
+>;
```

Comment:
Could you move these somewhere else? Maybe somewhere in a file with the name ApiMethodNameValidator. Might be a utilities, or types folder. 

Even tough the old appBridge had it in there, it creates a lot of noise and as it somehow all belongs together, might be easiest to put it somewhere away.